File: datastructures/queue.cpp
#include<iostream>

#define MAX_SIZE 10000

class Queue {
    private:
        int _front;
        int _rear;
        int _data[MAX_SIZE];

    public:
        Queue(){
            _rear = -1;
            _front = _rear;
        }

        bool isFull(){
            return (_rear >= MAX_SIZE - 1) ? true : false;
        }

        bool isEmpty(){
            return (_front >= _rear) ? true : false;
        }

        void enQueue(int data){
            if(!isFull()){
                _data[++_rear] = data;
            } else {
                throw "Queue is full";
            }
        }

        void deQueue(){
            if(!isEmpty()){
                _front++;
            } else {
                throw "Queue is empty";
            }
        }

        int peek(){
            if(!isEmpty()){
                return _data[_front+1];
            } else {
                throw "Queue is empty";
            }
        }
};

int main(){
    try {
        Queue queue;
        queue.enQueue(1);
        queue.enQueue(2);
        std::cout<<queue.peek()<<"\n";
        queue.deQueue();
        std::cout<<queue.peek()<<"\n";
        queue.deQueue();
        std::cout<<queue.peek()<<"\n";
    } catch (const char* msg) {
        std::cout<<msg<<"\n";
    }
}

File: datastructures/queue.js
class Node {
    constructor() {
        this.data = null;
        this.next = null;
    }
}

class Queue {
    constructor() {
        this.head = null;
    }

    isEmpty() {
        return this.head == null;
    }

    peek() {
        if(!this.isEmpty()) {
            return this.head.data;
        } else {
            throw 'Queue is empty';
        }
    }

    enQueue(value) {
        let tempNode = new Node();
        tempNode.data = value;
        if(this.isEmpty()) {
            this.head = tempNode;
        } else {
            let iterator = this.head;
            while(iterator.next != null) {
                iterator = iterator.next;
            }
            iterator.next = tempNode;
        }
    }

    deQueue() {
        if(!this.isEmpty()) {
            this.head = this.head.next;
        } else {
            throw 'Queue is empty';
        }
    }
}

let queue = new Queue();
queue.enQueue(10);
queue.enQueue(20);
queue.enQueue(30);
console.log(queue.peek());
queue.deQueue();
console.log(queue.peek());